### DISCUSSION TITLE
How to get Public Address on server from idToken with firebase custom auth

### DISCUSSION THREAD

**Discussion by yash-2523** [Web3Auth/Web3Auth#1158] [Help]
How to get the user public Address on server side for web3Auth-with-firebase setup. 

Like when decoded, I get payload, protectedHeader and key. But don't know how to get publicAddress from this. 

Thanks in advance.

**Comment by shahbaz17** [comment]
Hey @yash-2523 

Here, Is the server a backend server?

Then please take a look at https://web3auth.io/docs/sdk/node/usage#node-js-example 

So, you get a web3auth.provider and private key / public address in exchange for the idToken.

**Reply by yash-2523** [reply]
I get this when I decode the token on the backend.

```payload: {
    name: '',
    picture: '',
    iss: 'https://securetoken.google.com/web3auth-firebase',
    aud: 'web3auth-firebase',
    auth_time: 1672828344,
    user_id: '',
    sub: '',
    iat: 1672828344,
    exp: 1672831944,
    email: '',
    email_verified: ,
    firebase: { identities: [Object], sign_in_provider: 'google.com' }
  },
  protectedHeader: {
    alg: 'RS256',
    kid: '',
    typ: 'JWT'
  },
  key: PublicKeyObject {
    [Symbol(kKeyType)]: 'public',
    [Symbol(kAsymmetricKeyType)]: 'rsa'
  }```

So how to get the Ethereum public address from this?
